executor/sort.go

```diff
@@ -208,35 +263,25 @@ func (e *SortExec) fetchRowChunks(ctx context.Context) error {
 		if rowCount == 0 {
 			break
 		}
-		if e.alreadySpilled() {
-			// append chk to disk.
-			err := e.rowChunksInDisk.Add(chk)
-			if err != nil {
-				return err
-			}
-		} else {
-			e.rowChunks.Add(chk)
-			if atomic.LoadUint32(&e.exceeded) == 1 {
-				e.rowChunksInDisk, err = e.spillToDisk()
-				if err != nil {
-					return err
-				}
-				e.memTracker.Consume(-e.rowChunks.GetMemTracker().BytesConsumed())
-				e.rowChunks = nil // GC its internal chunks.
-				atomic.StoreUint32(&e.spilled, 1)
-			}
+		if err := e.rowChunks.Add(chk, e.generatePartition); err != nil {
```

Comment:
How about adding a new return value named `triggered bool` to indicate if it is spilled instead of using a callback function?
And the condition below `e.rowChunks.AlreadySpilled()` also can be replaced with `if triggered`.

Reply:
No. Before spilling disk, we need sort the partition data and save RowPtrs in e.Executor, so we need this callback function.